controllers/idler/idler_controller_test.go

```diff
@@ -682,51 +682,89 @@ func TestAppNameTypeForControllers(t *testing.T) {
 	}
 }
 
-func TestAppNameTypeForInidividualPods(t *testing.T) {
+func TestNotificationAppNameTypeForPods(t *testing.T) {
 	//given
 	idler := &toolchainv1alpha1.Idler{
 		ObjectMeta: metav1.ObjectMeta{
-			Name: "alex-stage",
+			Name: "feny-stage",
 			Labels: map[string]string{
-				toolchainv1alpha1.SpaceLabelKey: "alex",
+				toolchainv1alpha1.SpaceLabelKey: "feny",
 			},
 		},
 		Spec: toolchainv1alpha1.IdlerSpec{TimeoutSeconds: 60},
 	}
+	namespaces := []string{"dev", "stage"}
+	usernames := []string{"feny"}
+	nsTmplSet := newNSTmplSet(test.MemberOperatorNs, "feny", "advanced", "abcde11", namespaces, usernames)
+	mur := newMUR("feny")
 
-	t.Run("Test AppName/Type in notification", func(t *testing.T) {
-		namespaces := []string{"dev", "stage"}
-		usernames := []string{"alex"}
-		nsTmplSet := newNSTmplSet(test.MemberOperatorNs, "alex", "advanced", "abcde11", namespaces, usernames)
-		mur := newMUR("alex")
-		reconciler, req, cl, _ := prepareReconcile(t, idler.Name, getHostCluster, idler, nsTmplSet, mur)
-		idlerTimeoutPlusOneSecondAgo := time.Now().Add(-time.Duration(idler.Spec.TimeoutSeconds+1) * time.Second)
-		p := preparePayloadsSinglePod(t, reconciler, idler.Name, "todelete-", idlerTimeoutPlusOneSecondAgo).standalonePods[0]
-		// first reconcile to track pods
-		res, err := reconciler.Reconcile(context.TODO(), req)
-		assert.NoError(t, err)
-		assert.True(t, res.Requeue)
+	testpod := map[string]struct {
+		pcond           corev1.PodCondition
+		expectedAppType string
+	}{
+		"Individual-Completed-Pod": {
+			pcond:           corev1.PodCondition{Type: "Ready", Reason: "PodCompleted"},
+			expectedAppType: "Pod",
+		},
+		"Individual-NonCompleted-Pod": {
+			pcond:           corev1.PodCondition{Type: "Ready", Reason: ""},
+			expectedAppType: "Pod",
+		},
+		"Controlled-Completed-Pod": {
+			pcond:           corev1.PodCondition{Type: "Ready", Reason: "PodCompleted"},
+			expectedAppType: "Deployment",
+		},
+		"Controlled-NonCompleted-Pod": {
+			pcond:           corev1.PodCondition{Type: "Ready", Reason: ""},
+			expectedAppType: "Deployment",
+		},
```

Comment:
Let's cover two more use cases:
- There is no conditions at all.
- There are multiple conditions.

Reply:
@alexeykazakov the first "no conditions" i added, for 2nd "multiple condition" do you mean to pass a list of conditions or just another reason type.? 
My understanding is that pod can have a single condition status at a time. Not sure though

Reply:
I have added the multiple conditions too in test